polar-core/src/vm.rs

```diff
@@ -1291,8 +1297,7 @@ impl PolarVirtualMachine {
                     matches!(result.value(), Value::Variable(_)),
                     "Must have result variable as second arg."
                 );
-                let mut constructor = args.pop().unwrap();
-                constructor.map_replace(&mut |t| self.deref(t));
```

Comment:
Indeed. And for the record, the idea was to colocate all the `deep_deref`s with `QueryEvent`s. So that it (should) be obvious when a type is crossing the FFI boundary and needs to be derefed.

Reply:
We could actually encode this pretty cleanly with a newtype.

Reply:
SGTM